core-web/src/main/java/org/silverpeas/core/web/authentication/AuthenticationServlet.java

```diff
@@ -90,22 +93,35 @@
    * Ask for an authentication for the user behind the incoming HTTP request from a form.
    *
    * @param servletRequest the HTTP request.
-   * @param servletResponse the HTTP response.
+   * @param response the HTTP response.
    * @throws IOException when an error occurs while processing the request or sending the response.
    * @throws ServletException if the request for the POST couldn't be handled.
    */
   @Override
-  public void doPost(HttpServletRequest servletRequest, HttpServletResponse servletResponse)
+  public void doPost(HttpServletRequest servletRequest, HttpServletResponse response)
       throws IOException, ServletException {
     HttpRequest request = HttpRequest.decorate(servletRequest);
 
     final UserSessionStatus userSessionStatus = existOpenedUserSession(servletRequest);
+    final AuthenticationParameters authenticationParameters = new AuthenticationParameters(request);
     if (userSessionStatus.isValid()) {
+      final User connectedUser = userSessionStatus.getInfo().getUserDetail();
+      // Prevent trashing the user session on SSO authentication when user behind the current
+      // session is the one authenticated for SSO.
+      if (authenticationParameters.isSsoMode() &&
+          connectedUser.getLogin().equals(authenticationParameters.getLogin()) &&
+          connectedUser.getDomainId().equals(authenticationParameters.getDomainId())) {
+        SilverLogger.getLogger("silverpeas.sso").debug(() -> format(
+            "SSO Authentication of PRINCIPAL {0} on domain {1}, keeping existing user session alive {2}",
+            connectedUser.getLogin(), connectedUser.getDomainId(), userSessionStatus.getInfo().getSessionId()));
+        forward(request, response, "/Login");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a "try/catch" block for "forward". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1989)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a "try/catch" block for "forward". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1989)